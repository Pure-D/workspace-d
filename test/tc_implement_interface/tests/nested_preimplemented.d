module nested_preimplemented;

class HTMLElement : Element
{
	string name() @property
	{
		return "foo";
	}
}

abstract class Element : Node
{
	string tagName() @property
	{
		return "x";
	}

	abstract int numAttributes() @property;

	abstract string getAttribute(string name)
	{
		return name;
	}

	final void dontImplement() {}
}

interface Node
{
	void addChild(Node n);
	string name() @property;
	string tagName() @property;
}
