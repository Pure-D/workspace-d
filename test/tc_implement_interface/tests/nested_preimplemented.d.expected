implement 52
!string name()
!string tagName()
int numAttributes() @property
string getAttribute(string name)
return super.getAttribute(name);
!dontImplement
void addChild(Node n)